src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java

```diff
@@ -98,15 +114,25 @@ private UUID saveNewLetter(ILetterRequest letter, String messageId, String servi
             )
         );
 
-        // TODO: encrypt zip content
+        byte[] letterContent;
+        LocalDateTime createdAtTime = now();
+
+        if (isEncryptionEnabled) {
+            letterContent = encryptZipContents(letter, serviceName, id, zipContent, createdAtTime);
+        } else {
+            letterContent = zipContent;
+        }
```

Comment:
... then just set the contents here

-- edit --

oh and it becomes simpler: `encryptZipContents(dbLetter, zipContent)`. i guess replacing if/else to one liner is a bit too much, too big of a line